File: public/docs/_examples/dependency-injection/ts/app/heroes/hero-list.component.ts
// #docregion
import { Component }   from 'angular2/core';
import { Hero }        from './hero';
import { HeroService } from './hero.service';

@Component({
  selector: 'hero-list',
  template: `
  <div *ngFor="#hero of heroes">
    {{hero.id}} - {{hero.name}}
    ({{hero.isSecret ? 'secret' : 'public'}})
  </div>
  `,
})
export class HeroListComponent {
  heroes: Hero[];

  //#docregion ctor-signature
  constructor(heroService: HeroService) {
  //#enddocregion ctor-signature
    this.heroes = heroService.getHeroes();
  }
}

File: public/docs/_examples/dependency-injection/dart/lib/heroes/hero_list_component.dart
// #docregion
import 'package:angular2/core.dart';

import 'hero.dart';
import 'hero_service.dart';

@Component(
    selector: 'hero-list',
    template: '''
      <div *ngFor="let hero of heroes">
        {{hero.id}} - {{hero.name}}
        ({{hero.isSecret ? 'secret' : 'public'}})
      </div>''')
class HeroListComponent {
  final List<Hero> heroes;

  // #docregion ctor-signature
  HeroListComponent(HeroService heroService)
  // #enddocregion ctor-signature
    : heroes = heroService.getHeroes();
}
